python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -0,0 +1,209 @@
+
+
+def _get_session_data_location(base_url):
+    """
+    Returns the location of the session file on disk for a specific site.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_local_site_cache_location(base_url),
+        "authentication",
+        "login.ini"
+    )
+
+
+def delete_session_data(host):
+    """
+    Clears the session cache for the given site.
+
+    :param host: Site to clear the session cache for.
+    """
+    if not host:
+        logger.error("Current host not set, nothing to clear.")
+        return
+    logger.debug("Clearing session cached on disk.")
+    try:
+        info_path = _get_session_data_location(host)
+        if os.path.exists(info_path):
+            logger.debug("Session file found.")
+            os.remove(info_path)
+            logger.debug("Session cleared.")
+        else:
+            logger.debug("Session file not found: %s", info_path)
+    except:
+        logger.exception("Couldn't delete the session cache file!")
+
+
+def get_session_data(base_url):
+    """
+    Returns the cached login info if found.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Returns a dictionary with keys login and session_token or None
+    """
+    # Retrieve the location of the cached info
+    info_path = _get_session_data_location(base_url)
+    # Nothing was cached, return an empty dictionary.
+    if not os.path.exists(info_path):
+        logger.debug("No cache found at %s" % info_path)
+        return None
+    try:
+        # Read the login information
+        config = SafeConfigParser({"login": None, "session_token": None})
+        config.read(info_path)
+        if not config.has_section("LoginInfo"):
+            logger.debug("No Login info was found")
+            return None
+
+        login = config.get("LoginInfo", "login", raw=True)
+        session_token = config.get("LoginInfo", "session_token", raw=True)
+
+        if not login or not session_token:
+            logger.debug("Incomplete settings (login:%s, session_token:%s)" % (login, session_token))
+            return None
+
+        return {"login": login, "session_token": session_token}
+    except Exception:
+        logger.exception("Exception thrown while loading cached session info.")
+        return None
+
+
+def cache_session_data(host, login, session_token):
+    """
+    Caches the session data for a site and a user.
+
+    :param host: Site we want to cache a session for.
+    :param login: User we want to cache a session for.
+    :param session_token: Session token we want to cache.
+    """
+    # Retrieve the cached info file location from the host
+    info_path = _get_session_data_location(host)
+
+    # make sure the info_dir exists!
+    info_dir, info_file = os.path.split(info_path)
+    if not os.path.exists(info_dir):
+        os.makedirs(info_dir, 0700)
```

Comment:
is the umask set at this point as part of the calling code? If not, we should set it here.